libs/prefill/near_full_data_storage.py

```diff
@@ -0,0 +1,159 @@
+        current_usage_per = round(used_cap / total_cap * 100)
+
+        if memory_percent > current_usage_per:
+            # Get SNS configuration to retrieve available user data space
+            durability_values = DiskFailureRecoveryLib.retrieve_durability_values(master_obj, 'sns')
+            if not durability_values[0]:
+                LOGGER.error("Error in retrieving SNS values")
+                return durability_values
+            sns_values = {key: int(value) for key, value in durability_values[1].items()}
+            LOGGER.debug("Durability Values (SNS) %s", sns_values)
+            data_sns = sns_values['data']
+            sum_sns = sum(sns_values.values())
+            LOGGER.debug("Current usage : %s Expected disk usage : %s", current_usage_per,
+                         memory_percent)
+            write_percent = memory_percent - current_usage_per
+            expected_writes = (write_percent * total_cap) / 100
+            user_data_writes = data_sns / sum_sns * expected_writes
+            LOGGER.info("User writes to be performed %s bytes to attain %s full disk space",
+                        user_data_writes, memory_percent)
+            return True, user_data_writes
+        else:
+            LOGGER.info("Current Memory usage(%s) is already more than expected memory usage(%s)",
+                        current_usage_per, memory_percent)
+            return True, 0
+
+    @staticmethod
+    def perform_near_full_sys_writes(s3userinfo, user_data_writes, bucket_prefix: str,
+                                     client: int = 10) -> tuple:
+        """
+        Perform write operation till the memory is filled to given percentage
+        :param s3userinfo: S3user dictionary with access/secret key
+        :param user_data_writes: Write operation to be performed for specific bytes
+        :param bucket_prefix: Bucket prefix for the data written
+        :param client: number of clients
+        :return : list of dictionary
+                format : [{'bucket': bucket_name, 'obj_name_pref': obj_name, 'num_clients': client,
+                     'obj_size': obj_size, 'num_sample': sample}]
+        """
+        workload = [1, 16, 128, 256, 512]  # workload in mb
+        if CMN_CFG["setup_type"] == "HW":
+            workload.extend([1024, 2048, 3072, 4096])
+
+        mb = 1024 * 1024
+        workload = [each * mb for each in workload]  # convert to bytes
+        each_workload_byte = user_data_writes / len(workload)
+        return_list = []
+        for obj_size in workload:
+            samples = int(each_workload_byte / obj_size)
+            if samples > 0:
+                temp_client = client
+                if temp_client > samples:
+                    temp_client = samples
+                bucket_name = f'{bucket_prefix}-{obj_size}b'
+                obj_name = f'obj_{obj_size}'
+                resp = s3bench.s3bench(s3userinfo['accesskey'],
```

Comment:
You might want to use kwargs for long arguments function.

Reply:
fixed